### DISCUSSION TITLE
Remove Meteor checks for specific CPU architecture?

### DISCUSSION THREAD

**Discussion by xet7** [meteor/meteor#11661] [Feature Requests]
Hi,
Meteor has code to check for specific CPUs architecture, and it only allows installer to continue for specific architectures. For example, CPU checks like this https://github.com/Gatux/meteor/commit/877df973012c4a509b6d03fa833a0562e3e9a722 . 

For example, when I try to install new npm based Meteor installer, it stops like this:
```
$ sudo npm -g install meteor
...
found 0 vulnerabilities
npm ERR! code 1
npm ERR! path /usr/local/lib/node_modules/meteor
npm ERR! command failed
npm ERR! command sh -c node cli.js install
npm ERR! The current architecture is not supported: arm64

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/wekan/.npm/_logs/2021-10-01T16_20_34_570Z-debug.log
```

For Wekan https://wekan.github.io , first I build bundle at x64,  and then add fibers etc changes with these build scripts:
- [arm64 Linux](https://github.com/wekan/wekan/blob/master/releases/maintainer-make-bundle-a.sh), like RasPi3/RasPi4 and arm64 servers at Packet.com etc
- [s390x](https://github.com/wekan/wekan/blob/master/releases/maintainer-make-bundle-s.sh)
- [ppc64el](https://github.com/wekan/wekan/blob/master/releases/maintainer-make-bundle-o.sh)

Info about running bundle at those CPUs is at https://github.com/wekan/wekan/wiki/Raspberry-Pi

Can that CPU checking code be removed from Meteor? So that I could install Meteor and build directly on each CPU platform?

**Comment by xet7** [comment]
It seems there is Mac arm64 Node14 install possible with brew https://github.com/meteor/meteor/discussions/11633#discussioncomment-1349774 so if you build bundle on Rosetta x64 like with commands like this https://github.com/wekan/wekan/wiki/Mac and then use arm64 node14 and go to server directory https://github.com/wekan/wekan/blob/master/releases/maintainer-make-bundle-a.sh#L29 and install fibers https://github.com/wekan/wekan/blob/master/releases/maintainer-make-bundle-a.sh#L35 or in that fibers directory node build.js then it could work with newest Meteor, directy on Mac. But Docker depends, with OS is running inside of Docker.

Some running bundle info is at https://github.com/wekan/wekan/wiki/Raspberry-Pi for x64/arm64/s390x/ppc64el

There is node14 arm64 for Linux and Mac, but I don't know is there for Windows yet. It would maybe be possible to compile arm64 windows version of Node.js.

BTW, there is Docker Desktop kind of system from Rancher: https://news.ycombinator.com/item?id=28835690 .Seems to be for Windows/Mac/Linux.

I'm thinking, if Docker runs container that has Ubuntu arm64 version with Node.js and MongoDB and bundle, it should work.

I don't know is it possible to run macOS inside Docker container.

There is already versions of Node12, 14, 16 etc for arm64 Linux.

**Comment by filipenevola** [comment]
> Can that CPU checking code be removed from Meteor? So that I could install Meteor and build directly on each CPU platform?

I don't believe this is possible, otherwise, we will receive complaints all the time about Meteor not working as expected in a specific architecture, we need to be sure which architectures we can support.

You could do this in a fork, so you know that you are disabling checks and that maybe some features will be broken.

**Reply by xet7** [reply]
I tried to allow some more architectures at https://github.com/xet7/meteor/commit/17fd57a0ff5ad764441bf3b46fc6929b3cc77b37 and https://github.com/xet7/meteor/commit/67559dc1ca2b5b73bcb4af1583ebc8771f17ee78 but I got some syntax errors. If someone knows how to fix those syntax errors, help is very welcome.

**Reply by xet7** [reply]
If someone does not have some of those Linux platforms, it would be possible to test those for example with qemu:

```
$ sudo apt install qemu-system

$ ls /usr/bin/qemu-system*
/usr/bin/qemu-system-aarch64       /usr/bin/qemu-system-or1k
/usr/bin/qemu-system-alpha         /usr/bin/qemu-system-ppc
/usr/bin/qemu-system-arm           /usr/bin/qemu-system-ppc64
/usr/bin/qemu-system-cris          /usr/bin/qemu-system-ppc64le
/usr/bin/qemu-system-hppa          /usr/bin/qemu-system-riscv32
/usr/bin/qemu-system-i386          /usr/bin/qemu-system-riscv64
/usr/bin/qemu-system-lm32          /usr/bin/qemu-system-s390x
/usr/bin/qemu-system-m68k          /usr/bin/qemu-system-sh4
/usr/bin/qemu-system-microblaze    /usr/bin/qemu-system-sh4eb
/usr/bin/qemu-system-microblazeel  /usr/bin/qemu-system-sparc
/usr/bin/qemu-system-mips          /usr/bin/qemu-system-sparc64
/usr/bin/qemu-system-mips64        /usr/bin/qemu-system-tricore
/usr/bin/qemu-system-mips64el      /usr/bin/qemu-system-unicore32
/usr/bin/qemu-system-mipsel        /usr/bin/qemu-system-x86_64
/usr/bin/qemu-system-moxie         /usr/bin/qemu-system-xtensa
/usr/bin/qemu-system-nios2         /usr/bin/qemu-system-xtensaeb
```
If nodejs and mongodb is not in distro repos, it could be possible to compile those from source.
